src/main/java/domain/Car.java

```diff
@@ -0,0 +1,50 @@
+package domain;
+
+import dto.CarStatus;
+
+public class Car {
+
+    public static final int INITIAL_DISTANCE = 0;
+    public static final int MIN_NAME_LENGTH = 1;
+    public static final int MAX_NAME_LENGTH = 5;
+    public static final int FORWARD_NUMBER = 4;
+
+    private final RandomNumberGenerator randomNumberGenerator;
+    private final String name;
+
+    private int distance;
+
+    public Car(final RandomNumberGenerator randomNumberGenerator, final String name, final int distance) {
+        validateCarName(name);
+        this.randomNumberGenerator = randomNumberGenerator;
+        this.name = name;
+        this.distance = distance;
+    }
+
+    public Car(final RandomNumberGenerator randomNumberGenerator, final String name) {
+        this(randomNumberGenerator, name, INITIAL_DISTANCE);
+    }
+
+    private void validateCarName(final String name) {
+        final int length = name.length();
+        if (length < MIN_NAME_LENGTH || length > MAX_NAME_LENGTH) {
+            throw new IllegalArgumentException("이름 길이는 1이상 5이하의 문자열만 가능합니다.");
+        }
+    }
+
+    public void move() {
+        if (randomNumberGenerator == null) {
+            return;
+        }
```

Comment:
[질문] `randomNumberGenerator`가 `null`인 경우가 어떤 상황에 발생할 수 있나요?

Reply:
[코드를 짤 당시 생각]
자동차 객체를 테스트 할때 랜덤한 값을 테스트 하기가 어려웠어요.
그래서 인터페이스를 이용해 `randomNumberGenerator`를 생성자에서 주입받도록 했는데요.
테스트를 하다보니 모든 자동차에 `new RandomNumberGenerator()` 를 선언해 넣어주는게 불필요하다고 생각되어서 전진하는 테스트가 아닌 자동차에는 null을 넣어서 생성했어요.
따라서 move()를 실행할때 `randomNumberGenerator == null` 인 경우는 없을 것 같아요.
그렇지만 혹시나 null을 넣은 자동차에 `move()`를 호출하면 `NullPointerException`이 발생할 수도 있다 생각해서  넣어줬어요

[지금 생각]
테스트를 위해서 원래 코드를 수정했다는 생각이 들어요. 
그리고 자동차를 만들때 `randomNumberGenerator`에 왜 굳이 `null`을 넣었나 생각이 들어요.
```suggestion
    public void move() {
    
    
    
```
이렇게 변경하고 테스트의 자동차 생성자에 
`new RandomNumberGenerator()` 를 넣으려는데 어떻게 생각하시나요?

Reply:
제어할 수 없는 의존성을 바깥으로 빼는 것은 좋은 방법이죠 :)